File: app/src/main/java/com/icostel/arhitecturesample/ui/BackFragment.java
package com.icostel.arhitecturesample.ui;

import androidx.fragment.app.Fragment;

public abstract class BackFragment extends Fragment {
    private boolean shouldHandleBack = false;

    public void setShouldHandleBack(boolean shouldHandleBack) {
        this.shouldHandleBack = shouldHandleBack;
    }

    public boolean shouldHandleBack() {
        return shouldHandleBack;
    }

    public void onBackPress() { /* default nothing */ }
}

File: app/src/main/java/com/icostel/arhitecturesample/ui/BackFragment.kt
package com.icostel.arhitecturesample.ui

import androidx.fragment.app.Fragment

abstract class BackFragment : Fragment() {
    private var shouldHandleBack = false

    fun setShouldHandleBack(shouldHandleBack: Boolean) {
        this.shouldHandleBack = shouldHandleBack
    }

    fun shouldHandleBack(): Boolean {
        return shouldHandleBack
    }

    open fun onBackPress() { /* default nothing */
    }
}
